src/transformers/generation/candidate_generator.py

```diff
@@ -175,6 +184,7 @@ def get_candidates(self, input_ids: torch.LongTensor) -> Tuple[torch.LongTensor,
         # Don't generate more than `max_length - 1` candidates since the target model generates one extra token.
         new_cur_len = input_ids.shape[-1]
         max_new_tokens = min(int(self.num_assistant_tokens), self.generation_config.max_length - new_cur_len - 1)
+        min_new_tokens = min(max_new_tokens, max(self.min_length - new_cur_len, 0))
```

Comment:
```suggestion
        min_new_tokens = max(min(max_new_tokens, self.min_length - new_cur_len), 0)
```
We want a non-negative value [i.e. max(x, 0)] that is the minimum between `max_new_tokens` (so we don't force generate to go beyond it) and the minimum tokens set by generation config (so we don't generate fewer tokens than it).

Reply:
yeah, got lost in min maxes :)